CMakeLists.txt

```diff
@@ -0,0 +1,114 @@
+find_package(Threads REQUIRED)
+if(NOT THREADS_FOUND)
+	message(FATAL_ERROR "Threads not found")
+endif()
+
+include(GNUInstallDirs)
+
+################################################################################
+# Compiler flags:
+#   We want to use the same flags in the entire optee_client git
+################################################################################
+add_compile_options (
+	-Wall -Wbad-function-cast -Wcast-align
+	-Werror-implicit-function-declaration -Wextra
+	-Wfloat-equal -Wformat-nonliteral -Wformat-security
+	-Wformat=2 -Winit-self -Wmissing-declarations
+	-Wmissing-format-attribute -Wmissing-include-dirs
+	-Wmissing-noreturn -Wmissing-prototypes -Wnested-externs
+	-Wpointer-arith -Wshadow -Wstrict-prototypes
+	-Wswitch-default -Wunsafe-loop-optimizations
+	-Wwrite-strings -Werror -fPIC
+ 	-Wno-missing-field-initializers
+)
+
+find_program(CCACHE_FOUND ccache)
+if(CCACHE_FOUND)
+	set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
+	set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
+endif(CCACHE_FOUND)
+
+set (SRC
+	host/xtest/adbg/src/adbg_case.c
+	host/xtest/adbg/src/adbg_enum.c
+	host/xtest/adbg/src/adbg_expect.c
+	host/xtest/adbg/src/adbg_log.c
+	host/xtest/adbg/src/adbg_run.c
+	host/xtest/adbg/src/security_utils_hex.c
+	host/xtest/aes_perf.c
+	host/xtest/benchmark_1000.c
+	host/xtest/benchmark_2000.c
+	host/xtest/regression_1000.c
+	host/xtest/regression_4000.c
+	host/xtest/regression_5000.c
+	host/xtest/regression_6000.c
+	host/xtest/regression_7000.c
+	host/xtest/regression_8000.c
+	host/xtest/sha_perf.c
+	host/xtest/xtest_helpers.c
+	host/xtest/xtest_main.c
+	host/xtest/xtest_test.c
+)
+
+if (CFG_GP_SOCKETS)
+	list (APPEND SRC
+		host/xtest/regression_2000.c
+		host/xtest/sock_server.c
+		host/xtest/rand_stream.c
+	)
+endif()
+
+if (CFG_SECSTOR_TA_MGMT_PTA)
+	list (APPEND SRC host/xtest/install_ta.c)
+endif()
+
+if (CFG_SECURE_DATA_PATH)
+	list (APPEND SRC host/xtest/sdp_basic.c)
+endif()
+
+################################################################################
+# Built binary
+################################################################################
+add_executable (${PROJECT_NAME} ${SRC})
+
+target_compile_options (${PROJECT_NAME} PRIVATE -include conf.h)
+
+target_include_directories(${PROJECT_NAME}
+	PRIVATE host/xtest
+	PRIVATE host/xtest/adbg/include
+	PRIVATE host/xtest/xml/include
+	PRIVATE ${OPTEE_TEST_SDK}/host_include
+	PRIVATE ta/include
+	PRIVATE ta/aes_perf/include
+	PRIVATE ta/concurrent/include
+	PRIVATE ta/concurrent_large/include
+	PRIVATE ta/create_fail_test/include
+	PRIVATE ta/crypt/include
+	PRIVATE ta/enc_fs/include
+	PRIVATE ta/os_test/include
+	PRIVATE ta/rpc_test/include
+	PRIVATE ta/sdp_basic/include
+	PRIVATE ta/sha_perf/include
+	PRIVATE ta/sims/include
+	PRIVATE ta/socket/include
+	PRIVATE ta/storage_benchmark/include
+)
+
+target_link_libraries (${PROJECT_NAME}
+	PRIVATE pthread
```

Comment:
`${CMAKE_THREAD_LIBS_INIT}` instead of `pthread` (as you're using FindThreads on line 9)

BTW, we can implement the same cmake module-wrapper for `teec` (so we can use just cll `find_package(teec REQUIRED)`, which will automatically provide all paths in the same manner + this `FindTEEC.cmake` can be reused from CMakeLists.txt of other client applications)

Reply:
`find_package(teec REQUIRED)` would be nice, since there is no common (CMake) root for our gits.

Reply:
I'll replace `phtread` as you describe.
Can the rest wait until we've merged everything?

Reply:
> Can the rest wait until we've merged everything?

I think so.